core/src/main/java/feign/stream/StreamDecoder.java

```diff
@@ -35,35 +37,36 @@
  * <pre>
  * <code>
  * Feign.builder()
- *   .decoder(StreamDecoder.create(JacksonIteratorDecoder.create()))
+ *   .decoder(StreamDecoder.create(new Decoder.Default(), JacksonIteratorDecoder.create()))
  *   .doNotCloseAfterDecode() // Required for streaming
  *   .target(GitHub.class, "https://api.github.com");
  * interface GitHub {
  *  {@literal @}RequestLine("GET /repos/{owner}/{repo}/contributors")
  *   Stream<Contributor> contributors(@Param("owner") String owner, @Param("repo") String repo);
  * }</code>
  * </pre>
+ * 
+ * @author mroccyen
  */
 public final class StreamDecoder implements Decoder {
 
+  private final Decoder delegate;
   private final Decoder iteratorDecoder;
 
-  StreamDecoder(Decoder iteratorDecoder) {
+  StreamDecoder(Decoder delegate, Decoder iteratorDecoder) {
+    this.delegate = delegate;
     this.iteratorDecoder = iteratorDecoder;
   }
 
   @Override
   public Object decode(Response response, Type type)
       throws IOException, FeignException {
-    if (!(type instanceof ParameterizedType)) {
-      throw new IllegalArgumentException("StreamDecoder supports only stream: unknown " + type);
+    if (!isStream(type)) {
+      return delegate.decode(response, type);
```

Comment:
can't/shouldn't we use the one decoder to handle both?!

Seems confusing to me to have 2 decoders

Reply:
Thanks, I will try to modify it.